pkg/abi/linux/fuse.go

```diff
@@ -498,6 +545,63 @@ type FUSEWriteIn struct {
 	_ uint32
 }
 
+// SlicePayload represents a byte array which can be marshalled.
+//
+// +marshal dynamic
+type SlicePayload []byte
+
+// SizeBytes implements marshal.Marshallable.SizeBytes.
+func (s *SlicePayload) SizeBytes() int {
+	if s == nil {
+		return 0
+	}
+	return len(*s)
+}
+
+// MarshalBytes implements marshal.Marshallable.MarshalBytes.
+func (s *SlicePayload) MarshalBytes(dst []byte) []byte {
+	copy(dst, *s)
+	return dst[s.SizeBytes():]
+}
+
+// UnmarshalBytes implements marshal.Marshallable.UnmarshalBytes.
+func (s *SlicePayload) UnmarshalBytes(dst []byte) []byte {
+	panic("Unimplemented, SlicePayload is never unmarshalled")
+}
+
+// FUSEWritePayloadIn combines header - FUSEWriteIn and payload
+// in a single marshallable struct when sending request by the
+// kernel to the daemon
+//
+// +marshal dynamic
+type FUSEWritePayloadIn struct {
+	Header FUSEWriteIn
+	Payload SlicePayload
+}
+
+// SizeBytes implements marshal.Marshallable.SizeBytes.
+func (r *FUSEWritePayloadIn) SizeBytes() int {
+	if r == nil {
+		return 0 +
+			(*FUSEWriteIn)(nil).SizeBytes() +
+			(*SlicePayload)(nil).SizeBytes()
+	} else {
```

Comment:
Drop the else case. Because the if case returns, you don;t need an else. So we can save on indentation.